.gitignore

```diff
@@ -1,2 +1,2 @@
-./provider
+provider
```

Comment:
@gammazero this will ignore any _directory_ or file named `provider` recursively, which will include any untracked files added to [`cmd/provider`](https://github.com/filecoin-project/index-provider/tree/main/cmd/provider), hence the change to add `./` and make it explicit that the line here is meant to ignore the built binary.

=